warehouse/bigquery/bigquery.go

```diff
@@ -389,35 +389,29 @@ func (bq *HandleT) LoadTable(tableName string) error {
 	return err
 }
 
-func (bq *HandleT) MigrateSchema(diff warehouseutils.SchemaDiffT) (err error) {
-	if len(bq.Uploader.GetSchemaInWarehouse()) == 0 {
-		err = bq.createSchema()
-		if !checkAndIgnoreAlreadyExistError(err) {
-			return err
-		}
+func (bq *HandleT) CreateSchema() (err error) {
+	if len(bq.Uploader.GetSchemaInWarehouse()) > 0 {
+		return nil
 	}
-	processedTables := make(map[string]bool)
-	for _, tableName := range diff.Tables {
-		err = bq.createTable(tableName, diff.ColumnMaps[tableName])
+	err = bq.createSchema()
+	return err
+}
+
+func (bq *HandleT) MigrateTableSchema(tableName string, tableSchemaDiff warehouseutils.TableSchemaDiffT) (err error) {
+	if tableSchemaDiff.TableToBeCreated {
```

Comment:
This code is mostly duplicated across all warehouses.

Can we move them up to `upload.go`?

Add new functions to `whManager` interface -
```
CreateTable()
AddColumn()
```